src/lap_stats/lap_stats.c

```diff
@@ -297,86 +313,99 @@ static void sectorBoundaryEvent(const GpsSnapshot *gpsSnapshot)
     g_lastSectorTime = millis - g_lastSectorTimestamp;
     g_lastSectorTimestamp = millis;
     g_lastSector = g_sector;
-    ++g_sector;
+    g_at_sector = true;
+    update_sector_geo_circle(++g_sector);
 }
 
 /**
  * All logic associated with determining if we are at the finish line.
  */
-static void processFinishLogic(const GpsSnapshot *gpsSnapshot,
-                               const Track *track, const float targetRadius)
+static void process_finish_logic(const GpsSnapshot *gpsSnapshot)
 {
     if (!lapstats_lap_in_progress())
         return;
+
     if (!isGeoTriggerTripped(&g_finish_geo_trigger))
         return;
 
     const GeoPoint point = gpsSnapshot->sample.point;
-    const GeoPoint finishPoint = getFinishPoint(track);
-    const struct GeoCircle finishCircle = gc_createGeoCircle(finishPoint,
-                                          targetRadius);
-    if (!gc_isPointInGeoCircle(&point, finishCircle))
+    if (!gc_isPointInGeoCircle(&point, g_geo_circles.finish))
         return;
 
     // If we get here, then we have completed a lap.
     lap_finished_event(gpsSnapshot);
 }
 
+static void process_start_normal(const GpsSnapshot *gpsSnapshot)
+{
+        const GeoPoint point = gpsSnapshot->sample.point;
+        if (!gc_isPointInGeoCircle(&point, g_geo_circles.start))
+                return;
+
+        const tiny_millis_t time = gpsSnapshot->deltaFirstFix;
+        lap_started_event(time, &point, 0);
```

Comment:
???  This function is named this way so that we process start logic without the involvement of launch control.  Just because we are processing start logic doesn't imply that a start has happened (that is what the `lap_started_event` is for).  I could rename it to `process_start_no_launch_control` if that would make it clearer.